cratedb_toolkit/wtf/library.py

```diff
@@ -0,0 +1,587 @@
+                    sys.allocations;
+            """,
+            description="Table allocation across nodes, shards, and partitions.",
+        )
+
+        table_allocation_special = InfoElement(
+            name="table_allocation_special",
+            label="Table Allocations Special",
+            sql="""
+                SELECT decisions[2]['node_name'] AS node_name, COUNT(*) AS table_count
+                FROM sys.allocations
+                GROUP BY decisions[2]['node_name'];
+            """,
+            description="Table allocation. Special.",
+        )
+
+        table_shard_count = InfoElement(
+            name="table_shard_count",
+            label="Table Shard Count",
+            sql="""
+                SELECT
+                    table_schema,
+                    table_name,
+                    SUM(number_of_shards) AS num_shards
+                FROM
+                    information_schema.table_partitions
+                WHERE
+                    closed = false
+                GROUP BY table_schema, table_name;
+            """,
+            description="Total number of shards per table.",
+        )
+
+        total_count = InfoElement(
+            name="shard_total_count",
+            label="Number of shards",
+            description="Total number of shards.",
+            sql="""
+                SELECT COUNT(*) AS shard_count
+                FROM sys.shards
+            """,
+            transform=get_single_value("shard_count"),
+        )
+
+        # TODO: Are both `translog_uncommitted` items sensible?
+        translog_uncommitted = InfoElement(
+            name="translog_uncommitted",
+            label="Uncommitted Translog",
+            description="Check if translogs are committed properly by comparing the "
+            "`flush_threshold_size` with the `uncommitted_size` of a shard.",
+            sql="""
+                SELECT
+                  sh.table_name,
+                  sh.partition_ident,
+                  SUM(sh.translog_stats['uncommitted_size']) / POWER(1024, 3) as "translog_uncomitted_in_gib"
+                FROM information_schema.table_partitions tp
+                JOIN sys.shards sh USING (table_name, partition_ident)
+                WHERE sh.translog_stats['uncommitted_size'] > settings['translog']['flush_threshold_size']
+                GROUP BY 1, 2
+                ORDER BY 3 DESC;
+            """,
+        )
+        translog_uncommitted_size = InfoElement(
+            name="translog_uncommitted_size",
+            label="Total uncommitted translog size",
+            description="A large number of uncommitted translog operations can indicate issues with shard replication.",
+            sql="""
+                SELECT COALESCE(SUM(translog_stats['uncommitted_size']), 0) AS translog_uncommitted_size
+                FROM sys.shards;
+            """,
+            transform=get_single_value("translog_uncommitted_size"),
+            unit="bytes",
+        )
```

Comment:
Isn't the query above more detailed? I think this one can be skipped...

Reply:
See also GH-169.